addons/mail/controllers/discuss/binary.py

```diff
@@ -92,3 +61,14 @@ def fetch_image(self, channel_id, attachment_id, width=0, height=0, **kwargs):
             ._get_image_stream_from(attachment_sudo, width=int(width), height=int(height))
             .get_response(as_attachment=kwargs.get("download"))
         )
+
+    @add_guest_to_context
+    @http.route()
```

Comment:
`http.route` should always be the outter most decorator (otherwise it registers the route handler before the other decorator is applied)